datafusion/sql/src/unparser/expr.rs

```diff
@@ -582,6 +582,47 @@ impl Unparser<'_> {
                     expr: Box::new(date_expr),
                 });
             }
+        } else if func_name.to_lowercase() == "date_part"
```

Comment:
As we have more variants for `date_part` now  it is probably better to include this as match inside `if func_name.to_lowercase() == "date_part"` with comment that `DateFieldExtractStyle.DatePart` does not require override